csv-core/src/reader.rs

```diff
@@ -666,7 +666,9 @@ impl Reader {
         let mut state = self.dfa_state;
         while nin < input.len() && nout < output.len() && nend < ends.len() {
             let (s, has_out) = self.dfa.get_output(state, input[nin]);
-            self.line += (input[nin] == b'\n') as u64;
+            if input[nin] == b'\n'{
+                self.line += 1;
+            }
```

Comment:
Do you have benchmarks backing this up?  I'm a little skeptical that the cost of looking up `self.line` in order to increment it by zero is higher than the cost of adding a branch in the code path, at least on most modern CPUs where branches do unpleasant things to instruction pipelining.